stl/inc/memory

```diff
@@ -3872,7 +3873,7 @@ protected:
         _Ptr.notify_all();
     }
 
-    atomic<_Ty*> _Ptr{nullptr};
+    atomic<remove_extent_t<_Ty>*> _Ptr{nullptr};

```

Comment:
No change requested: Regarding the ABI implications that @strega-nil-ms mentioned in https://github.com/microsoft/STL/pull/1339#pullrequestreview-961181700 - while changing data members is always concerning from an ABI perspective, I agree that the actual bits being stored are the same. Additionally, the previous code would generally fail to compile for simple cases like `store()`, which indicates that we shouldn't have to worry about mismatch issues. (The one time we can make unrestricted changes to layout is when the code couldn't compile at all before - this is not quite "wouldn't compile at all" but close.)